timemachine/cpp/src/kernels/k_harmonic_angle.cuh

```diff
@@ -58,35 +56,33 @@ void __global__ k_harmonic_angle_inference(
 
     RealType delta = top / nijk - cos(a0);
 
-    RealType prefactor = lambda_offset[a_idx] + lambda_mult[a_idx] * lambda;
-
     if (du_dx) {
         for (int d = 0; d < 3; d++) {
             RealType grad_i = ka * delta * (rij[d] * top / (n3ij * njk) + (-rjk[d]) / nijk);
-            atomicAdd(du_dx + i_idx * D + d, FLOAT_TO_FIXED_BONDED<RealType>(grad_i * prefactor));
+            atomicAdd(du_dx + i_idx * D + d, FLOAT_TO_FIXED_BONDED<RealType>(grad_i));
 
             RealType grad_j =
                 ka * delta *
                 ((-rij[d] * top / (n3ij * njk) + (-rjk[d]) * top / (nij * n3jk) + (rij[d] + rjk[d]) / nijk));
-            atomicAdd(du_dx + j_idx * D + d, FLOAT_TO_FIXED_BONDED<RealType>(grad_j * prefactor));
+            atomicAdd(du_dx + j_idx * D + d, FLOAT_TO_FIXED_BONDED<RealType>(grad_j));
 
             RealType grad_k = ka * delta * (-rij[d] / nijk + rjk[d] * top / (nij * n3jk));
-            atomicAdd(du_dx + k_idx * D + d, FLOAT_TO_FIXED_BONDED<RealType>(grad_k * prefactor));
+            atomicAdd(du_dx + k_idx * D + d, FLOAT_TO_FIXED_BONDED<RealType>(grad_k));
         }
     }
 
     if (du_dp) {
         RealType dka_grad = delta * delta / 2;
-        atomicAdd(du_dp + ka_idx, FLOAT_TO_FIXED_BONDED(dka_grad * prefactor));
+        atomicAdd(du_dp + ka_idx, FLOAT_TO_FIXED_BONDED(dka_grad));
         RealType da0_grad = delta * ka * sin(a0);
-        atomicAdd(du_dp + a0_idx, FLOAT_TO_FIXED_BONDED(da0_grad * prefactor));
+        atomicAdd(du_dp + a0_idx, FLOAT_TO_FIXED_BONDED(da0_grad));
     }
 
     if (u) {
-        atomicAdd(u + i_idx, FLOAT_TO_FIXED_BONDED(ka / 2 * delta * delta * prefactor));
+        atomicAdd(u + i_idx, FLOAT_TO_FIXED_BONDED(ka / 2 * delta * delta));
     }
 
     if (du_dl) {
-        atomicAdd(du_dl + i_idx, FLOAT_TO_FIXED_BONDED(lambda_mult[a_idx] * ka / 2 * delta * delta));
+        atomicAdd(du_dl + i_idx, FLOAT_TO_FIXED_BONDED(0.0));
```

Comment:
nit: is there a reason to leave this in (assuming it's a no-op)?

Reply:
Good call -- removed from here (and `k_harmonic_bond`, `k_periodic_torsion`) in https://github.com/proteneer/timemachine/pull/891/commits/c60addb402e3d8a9a196ef9bb0795f912583ca6c